src/UI/Root/Components/Header/EnvSelector/EnvironmentSelector.test.tsx

```diff
@@ -1,13 +1,45 @@
 import React, { act } from "react";
 import { MemoryRouter } from "react-router";
 import { Router } from "react-router-dom";
-import { render, screen } from "@testing-library/react";
+import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
+import { render, screen, waitFor } from "@testing-library/react";
 import { userEvent } from "@testing-library/user-event";
 import { createMemoryHistory } from "history";
-import { RemoteData } from "@/Core";
-import { Environment, dependencies } from "@/Test";
+import { HttpResponse, http } from "msw";
+import { setupServer } from "msw/node";
+import { FlatEnvironment, RemoteData } from "@/Core";
+import { AuthProvider, KeycloakAuthConfig, LocalConfig } from "@/Data";
+import { AuthTestWrapper, Environment, dependencies } from "@/Test";
 import { DependencyProvider } from "@/UI/Dependency";
+import ErrorBoundary from "@/UI/Utils/ErrorBoundary";
 import { EnvSelectorWithData as EnvironmentSelector } from "./EnvSelectorWithData";
+import { EnvironmentSelectorItem } from "./EnvSelectorWrapper";
+
+const setup = (
```

Comment:
There are arguments that can have defaults to avoid ending up with so many anonymous arguments down in the tests. And if you organise the order a bit, you'd have minimal arguments to pass to override the variables.

```ts
const setup = (
 // I placed this one first since it's the one that changes the most.  also, avoid passing it as an anonymous function
// it makes it hard to read months later. 
  callback?: (item: EnvironmentSelectorItem) => void,
  // Default it to undefined, this avoids the need to pass it as undefined when calling the method.
// I think you can omit setting the default altogether if the argument is optional... not sure right now, but you get the drift.
  config?: KeycloakAuthConfig | LocalConfig = undefined,
  // these two below are always the same, so you could consider removing them from the arg list and have them declared in your method directly. Or have defaults like these if you feel like they might change or be overridden.
  environments?: FlatEnvironment[] = Environment.filterable,  
  selectedEnv?: FlatEnvironment | undefined = envA, 
) => {
  const queryClient = new QueryClient();
  return (
    <ErrorBoundary>
      <MemoryRouter>
        <QueryClientProvider client={queryClient}>
          <AuthProvider config={config}>
            <AuthTestWrapper dependencies={dependencies}>
              <EnvironmentSelector
                environments={RemoteData.success(environments)}
                onSelectEnvironment={callback || () => {}}
                selectedEnvironment={selectedEnv}
              />
            </AuthTestWrapper>
          </AuthProvider>
        </QueryClientProvider>
      </MemoryRouter>
    </ErrorBoundary>
  );
};
```

Reply:
changed

Reply:
sorry if I'm being annoying, but callback doesn't mean much on what it's a callback for. Can you adjust the naming of the method? onSelectEnvironment or something similar?
 Can you also name the anonymous methods in the places where you use that argument ?

Reply:
No worries, you are not! renamed and added names for callback functions